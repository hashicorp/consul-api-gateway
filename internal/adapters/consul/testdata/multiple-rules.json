{
  "Name": "multiple-rules",
  "Namespace": "k8s",
  "Hostnames": [
    "example.com"
  ],
  "Rules": [
    {
      "Matches": [
        {
          "Path": {
            "Type": "HTTPPathMatchExact",
            "Value": "/prefix-1"
          }
        },
        {
          "Path": {
            "Type": "HTTPPathMatchExact",
            "Value": "/other-1"
          }
        }
      ],
      "Filters": [
        {
          "Type": "HTTPHeaderFilter",
          "Header": {
            "Set": {
              "x-set": "1"
            },
            "Add": {
              "x-add": "2"
            },
            "Remove": [
              "x-remove"
            ]
          }
        },
        {
          "Type": "HTTPHeaderFilter",
          "Header": {
            "Set": {
              "x-set-too": "1"
            },
            "Add": {
              "x-add-too": "2"
            }
          }
        }
      ],
      "Services": [
        {
          "Service": {
            "ConsulNamespace": "namespace",
            "Service": "service"
          },
          "Weight": 1,
          "Filters": [
            {
              "Type": "HTTPHeaderFilter",
              "Header": {
                "Set": {
                  "x-set": "3"
                },
                "Add": {
                  "x-add": "4"
                },
                "Remove": [
                  "x-remove-a"
                ]
              }
            }
          ]
        },
        {
          "Service": {
            "ConsulNamespace": "namespace",
            "Service": "another-service"
          },
          "Weight": 1,
          "Filters": [
            {
              "Type": "HTTPHeaderFilter",
              "Header": {
                "Add": {
                  "x-add": "4"
                },
                "Remove": [
                  "x-remove-b"
                ]
              }
            }
          ]
        }
      ]
    },
    {
      "Matches": [
        {
          "Path": {
            "Type": "HTTPPathMatchExact",
            "Value": "/prefix-2"
          }
        }
      ],
      "Filters": [
        {
          "Type": "HTTPHeaderFilter",
          "Header": {
            "Set": {
              "x-set-2": "5"
            },
            "Add": {
              "x-add-2": "6"
            },
            "Remove": [
              "x-remove-2"
            ]
          }
        },
        {
          "Type": "HTTPHeaderFilter",
          "Header": {
            "Set": {
              "x-set-2-too": "7"
            },
            "Add": {
              "x-add-2-too": "8"
            }
          }
        }
      ],
      "Services": [
        {
          "Service": {
            "ConsulNamespace": "namespace",
            "Service": "service"
          },
          "Weight": 1,
          "Filters": [
            {
              "Type": "HTTPHeaderFilter",
              "Header": {
                "Set": {
                  "x-set-2": "6"
                },
                "Add": {
                  "x-add-2": "7"
                },
                "Remove": [
                  "x-remove-a"
                ]
              }
            }
          ]
        },
        {
          "Service": {
            "ConsulNamespace": "namespace",
            "Service": "another-service"
          },
          "Weight": 1,
          "Filters": [
            {
              "Type": "HTTPHeaderFilter",
              "Header": {
                "Add": {
                  "x-add": "4"
                },
                "Remove": [
                  "x-remove-b"
                ]
              }
            }
          ]
        }
      ]
    }
  ]
}