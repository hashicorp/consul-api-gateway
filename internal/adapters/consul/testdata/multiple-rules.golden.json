{
  "Router": {
    "Kind": "service-router",
    "Name": "multiple-rules",
    "Namespace": "k8s",
    "Routes": [
      {
        "Match": {
          "HTTP": {
            "PathExact": "/prefix-1"
          }
        },
        "Destination": {
          "Service": "multiple-rules-0",
          "Namespace": "k8s",
          "RequestHeaders": {
            "Add": {
              "x-add": "2",
              "x-add-too": "2"
            },
            "Set": {
              "x-set": "1",
              "x-set-too": "1"
            },
            "Remove": [
              "x-remove"
            ]
          }
        }
      },
      {
        "Match": {
          "HTTP": {
            "PathExact": "/other-1"
          }
        },
        "Destination": {
          "Service": "multiple-rules-0",
          "Namespace": "k8s",
          "RequestHeaders": {
            "Add": {
              "x-add": "2",
              "x-add-too": "2"
            },
            "Set": {
              "x-set": "1",
              "x-set-too": "1"
            },
            "Remove": [
              "x-remove"
            ]
          }
        }
      },
      {
        "Match": {
          "HTTP": {
            "PathExact": "/prefix-2"
          }
        },
        "Destination": {
          "Service": "multiple-rules-1",
          "Namespace": "k8s",
          "RequestHeaders": {
            "Add": {
              "x-add-2": "6",
              "x-add-2-too": "8"
            },
            "Set": {
              "x-set-2": "5",
              "x-set-2-too": "7"
            },
            "Remove": [
              "x-remove-2"
            ]
          }
        }
      }
    ],
    "CreateIndex": 0,
    "ModifyIndex": 0
  },
  "Splitters": [
    {
      "Kind": "service-splitter",
      "Name": "multiple-rules-0",
      "Namespace": "k8s",
      "Splits": [
        {
          "Weight": 50,
          "Service": "service",
          "Namespace": "namespace",
          "RequestHeaders": {
            "Add": {
              "x-add": "4"
            },
            "Set": {
              "x-set": "3"
            },
            "Remove": [
              "x-remove-a"
            ]
          }
        },
        {
          "Weight": 50,
          "Service": "another-service",
          "Namespace": "namespace",
          "RequestHeaders": {
            "Add": {
              "x-add": "4"
            },
            "Remove": [
              "x-remove-b"
            ]
          }
        }
      ],
      "CreateIndex": 0,
      "ModifyIndex": 0
    },
    {
      "Kind": "service-splitter",
      "Name": "multiple-rules-1",
      "Namespace": "k8s",
      "Splits": [
        {
          "Weight": 50,
          "Service": "service",
          "Namespace": "namespace",
          "RequestHeaders": {
            "Add": {
              "x-add-2": "7"
            },
            "Set": {
              "x-set-2": "6"
            },
            "Remove": [
              "x-remove-a"
            ]
          }
        },
        {
          "Weight": 50,
          "Service": "another-service",
          "Namespace": "namespace",
          "RequestHeaders": {
            "Add": {
              "x-add": "4"
            },
            "Remove": [
              "x-remove-b"
            ]
          }
        }
      ],
      "CreateIndex": 0,
      "ModifyIndex": 0
    }
  ]
}